### PR TITLE
[Merged by Bors] - feat: Port CategoryTheory.Monoidal.NaturalTransformation

### PR DESCRIPTION
**Pull request by adamtopaz**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/CategoryTheory/Monoidal/NaturalTransformation.lean [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+instance (F : MonoidalFunctor C D) [IsEquivalence F.toFunctor] : IsIso (monoidalUnit F) :=
+  haveI : ∀ X : C, IsIso ((monoidalUnit F).toNatTrans.app X) := by
+    intros
+    dsimp
+    infer_instance
+  MonoidalNatIso.isIso_of_isIso_app _
+
+/-- The counit of a monoidal equivalence can be upgraded to a monoidal natural transformation. -/
+@[simp] -- Porting note: changed to simp
```
Comment (ChrisHughes24) [inline Mathlib/CategoryTheory/Monoidal/NaturalTransformation.lean:226]: I don't think this is the right thing to do here. `simp` does not do the same thing as `simps`. I think it is the `let` that is stopping `simps` from working here, so either `simps` should be changed to work with `let`s or the definition should be rewritten without `let` and with a porting note explaining the change

Comment (adamtopaz) [reply Mathlib/CategoryTheory/Monoidal/NaturalTransformation.lean]: Hmmm.... it seems that the `let` was not the issue. Something else is going on. Will investigate soon.

Comment (adamtopaz) [reply Mathlib/CategoryTheory/Monoidal/NaturalTransformation.lean]: Both `[simp]` have been reverted to `simps!` and the errors fixed by specifying the toNatTrans projection.

**Pull request closed without merging**